x/protocolpool/keeper/keeper_test.go

```diff
@@ -176,78 +111,421 @@ func (suite *KeeperTestSuite) TestGetCommunityPool() {
-		Recipient:  addrStr,
-		Percentage: math.LegacyMustNewDecFromStr("0.3"),
-		Expiry:     nil,
+		funds, err := suite.poolKeeper.GetAllContinuousFunds(suite.ctx)
+		suite.Require().NoError(err)
+		suite.Require().Len(funds, 1, "expected one continuous fund in store")
+		suite.Require().Equal(fund.Recipient, funds[0].Recipient)
+		suite.Require().Equal(fund.Percentage, funds[0].Percentage)
+		suite.Require().Equal(fund.Expiry, funds[0].Expiry)
 	})
-	suite.Require().NoError(err)
 
-	err = suite.poolKeeper.SetToDistribute(suite.ctx)
-	suite.Require().NoError(err)
+	suite.Run("many funds in store", func() {
+		suite.SetupTest()
 
-	// Verify that LastBalance was set correctly
-	lastBalance, err := suite.poolKeeper.LastBalance.Get(suite.ctx)
-	suite.Require().NoError(err)
-	suite.Require().Equal(sdk.NewCoins(sdk.NewCoin("stake", math.NewInt(1000000)), sdk.NewCoin("foo", math.NewInt(1234))), lastBalance.Amount)
+		totalFunds := 10
 
-	// Verify that a distribution was set
-	var distribution types.DistributionAmount
-	err = suite.poolKeeper.Distributions.Walk(suite.ctx, nil, func(key time.Time, value types.DistributionAmount) (bool, error) {
-		distribution = value
-		return true, nil
-	})
-	suite.Require().NoError(err)
-	suite.Require().Equal(sdk.NewCoins(sdk.NewCoin("stake", math.NewInt(1000000)), sdk.NewCoin("foo", math.NewInt(1234))), distribution.Amount)
+		// Insert a number of funds.
+		for i := 0; i < totalFunds; i++ {
+			accAddr := sdk.AccAddress(fmt.Sprintf("ao%d__________________", i))
 
-	// Test case when balance is zero
-	zeroBal := sdk.NewCoin(sdk.DefaultBondDenom, math.ZeroInt())
-	suite.bankKeeper.EXPECT().GetBalance(suite.ctx, poolDistrAcc.GetAddress(), sdk.DefaultBondDenom).Return(zeroBal)
-	suite.bankKeeper.EXPECT().GetBalance(suite.ctx, poolDistrAcc.GetAddress(), "foo").Return(sdk.NewCoin("foo", math.ZeroInt()))
+			fund := types.ContinuousFund{
+				Recipient:  accAddr.String(),
+				Percentage: math.LegacyMustNewDecFromStr("0.1"),
+				Expiry:     nil,
+			}
+			err := suite.poolKeeper.ContinuousFunds.Set(suite.ctx, accAddr, fund)
+			suite.Require().NoError(err)
+		}
 
-	err = suite.poolKeeper.SetToDistribute(suite.ctx)
-	suite.Require().NoError(err)
+		funds, err := suite.poolKeeper.GetAllContinuousFunds(suite.ctx)
+		suite.Require().NoError(err)
+		suite.Require().Len(funds, totalFunds, "expected many continuous funds in store")
 
-	// Verify that no new distribution was set
-	count := 0
-	err = suite.poolKeeper.Distributions.Walk(suite.ctx, nil, func(key time.Time, value types.DistributionAmount) (bool, error) {
-		count++
-		return false, nil
+		// verify each inserted fund's percentage.
+		for _, f := range funds {
+			suite.Require().Equal(math.LegacyMustNewDecFromStr("0.1"), f.Percentage)
+		}
 	})
-	suite.Require().NoError(err)
-	suite.Require().Equal(1, count) // Only the previous distribution should exist
+}
+
+func (suite *KeeperTestSuite) TestDistributeFunds() {
```

Comment:
this test is a little hard to follow, and seems like it would be cumbersome to add test cases in the future. it would be nice to have a test case where you can define:

- params
- initial balance of the pool
- a list of ContinuousFunds
- expected balances

Reply:
I've simplified as much as i could here:

[update](https://github.com/cosmos/cosmos-sdk/pull/24116/commits/c7bd251a45274c6e7bea2f610435144a49ab313c)

A lot of this setup and initialization logic is tied to each other with the percentages and expected stream amounts.   This gives you `params` and `initialBalance` easily set-able at least